### PR TITLE
[PIR] set lr scheduler in pir

### PR DESCRIPTION
**Pull request by kangguangli**
### PR types

Others
### PR changes

Others
### Description

set lr scheduler in pir

#### Others
Pcard-67164

### CODE REVIEW

File: python/paddle/optimizer/optimizer.py [hunk 1]
```diff
@@ -453,29 +454,63 @@ def do_create():
...
+                            paramete_meta, startup_program.global_block()
+                        )
+                        init_result.persistable = True
+                        set_parameter(init_result, lr_name)
+                    main_program.move_parameters_from(startup_program)
+
+                    if not isinstance(lr_var, paddle.pir.OpResult):
+                        self._learning_rate._var_name = lr_name
+                        with paddle.static.program_guard(main_program):
+                            param = get_parameter(lr_name, _lr_dtype, [])
```
Comment (YuanRisheng) [inline python/paddle/optimizer/optimizer.py:481]: 这个再封装一下感觉怎么样，估计get_parameter这种少不了使用，每次写guard不太方便，搞一个get_param_from_program如何

**Review by YuanRisheng — APPROVED** [review]

File: python/paddle/optimizer/optimizer.py [hunk 2]
```diff
@@ -453,29 +454,63 @@ def do_create():
...
+                        )
+                        paramete_meta = paddle.pir.core.ParameterMeta(
+                            [], _lr_dtype
+                        )
+                        init_result = initializer(
+                            paramete_meta, startup_program.global_block()
+                        )
+                        init_result.persistable = True
+                        set_parameter(init_result, lr_name)
+                    main_program.move_parameters_from(startup_program)
```
Comment (YuanRisheng) [inline python/paddle/optimizer/optimizer.py:476]: 这段代码不太容易理解，后边类似代码预期还会写不少，这个封装成函数会不会好一些，比如叫init_global_value？之类的怎么样

File: python/paddle/optimizer/optimizer.py [hunk 3]
```diff
@@ -727,7 +762,10 @@ def _global_learning_rate(self, program=None):
         :return:
         """
         if program is None:
-            program = framework.default_main_program()
+            if in_dygraph_mode():
+                program = framework.default_main_program()
+            else:
+                program = paddle.static.default_main_program()
```
Comment (0x45f) [inline python/paddle/optimizer/optimizer.py:768]: 这里统一用`program = paddle.static.default_main_program()`应该就可以吧

**Review by 0x45f — APPROVED** [review]

File: python/paddle/optimizer/optimizer.py (hunk 3 above)
Comment (kangguangli) [reply python/paddle/optimizer/optimizer.py:768]: 不行，会影响动态图下的学习率设置。

File: python/paddle/optimizer/optimizer.py (hunk 2 above)
Comment (kangguangli) [reply python/paddle/optimizer/optimizer.py:476]: 确实可以，不过我觉得接口形态目前不好确定，可以等后面有类似需求再抽象一下。

File: python/paddle/optimizer/optimizer.py (hunk 1 above)
Comment (kangguangli) [reply python/paddle/optimizer/optimizer.py:481]: 同上。
